statement_test.go

```diff
@@ -844,3 +844,21 @@ func TestDriverValuer(t *testing.T) {
 	require.Error(t, err, "[]uuid.UUID should fail without driver.Valuer")
 	require.Contains(t, err.Error(), "unsupported data type: UUID")
 }
+
+func TestMixedTypeSliceBinding(t *testing.T) {
+	db := openDbWrapper(t, ``)
+	defer closeDbWrapper(t, db)
+
+	createTable(t, db, `CREATE TABLE mixed_slice_test (foo TEXT)`)
+	_, err := db.Exec(`INSERT INTO mixed_slice_test VALUES ('hello this is text'), ('other')`)
+	require.NoError(t, err)
+
+	mixedSlice := []any{"hello this is text", 2}
+
+	_, err = db.Query("FROM mixed_slice_test WHERE foo IN ?", mixedSlice)
+	require.ErrorContains(t, err, "mixed types in slice at index 0: cannot bind BIGINT and VARCHAR together: index: 1")
```

Comment:
nit: maybe we can change this to `mixed types in slice: cannot bind BIGINT (index 1) and VARCHAR (index 0)`? 🤔